src/utils/rocksdb_serialization.hpp

```diff
@@ -30,6 +30,50 @@ namespace memgraph::utils {
 static constexpr const char *outEdgeDirection = "0";
 static constexpr const char *inEdgeDirection = "1";
 
+namespace {
+struct StartEndPositions {
+  size_t start;
+  size_t end;
+
+  size_t Size() const { return end - start; }
+  bool Valid() const { return start != std::string::npos; }
+};
+
+inline StartEndPositions FindPartStartEndPositions(const std::string_view str, const char delim,
+                                                   unsigned int partNumber) {
+  StartEndPositions result{0, 0};
+  for (int i = 0; i < partNumber; ++i) {
+    result.start = result.end;
+    result.end = str.find(delim, result.start);
+    if (result.end == std::string::npos) {
+      if (i < partNumber - 1) {
+        // We didn't find enough parts.
+        result.start = std::string::npos;
+      }
+      return result;
+    }
+    ++result.end;
```

Comment:
On calls where we are not exhausting the number of parts, this will cause the delimiter to be included in the final result.
Is that what we wanted?
FindPartStartEndPositions("A|B|C", '|', 1) -> "A|"

Reply:
In this case result.end == std::string::npos, so it's end of a string + 1.
At the end of loop we increment result.end for the case when '|' is found in middle of string

Reply:
The ++end is being called outside the `if (result.end == std::string::npos) {...}` block.
Here's a simple test I ran: https://godbolt.org/z/bz493W8Kf
I print the results using find and split.
P.S.
Make sure godbolt is executing the code, you should see the terminal and the results there.

Reply:
Thank you, Andreja! You are right :) I'm adding more unit tests for extraction functions.